include/watchdog.h

```diff
@@ -1,116 +1,168 @@
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-#define WDT_MODE		(BIT(1))
-#define WDT_MODE_OFFSET         (1)
-#define WDT_TIMEOUT_MASK        (0xF)
-
-enum wdt_mode {
-	WDT_MODE_RESET = 0,
-	WDT_MODE_INTERRUPT_RESET
-};
+/**
+ * @brief Watchdog operational mode.
+ */
+#define WDT_OP_MODE_RESET		(0)
+#define WDT_OP_MODE_INTERRUPT_RESET	BIT(0)
+#define WDT_OP_MODE_MASK		(0x1)
 
 /**
- * WDT clock cycles for timeout type.
+ * @brief Watchdog behavior when CPU is in sleep state.
  */
-enum wdt_clock_timeout_cycles {
-	WDT_2_16_CYCLES,
-	WDT_2_17_CYCLES,
-	WDT_2_18_CYCLES,
-	WDT_2_19_CYCLES,
-	WDT_2_20_CYCLES,
-	WDT_2_21_CYCLES,
-	WDT_2_22_CYCLES,
-	WDT_2_23_CYCLES,
-	WDT_2_24_CYCLES,
-	WDT_2_25_CYCLES,
-	WDT_2_26_CYCLES,
-	WDT_2_27_CYCLES,
-	WDT_2_28_CYCLES,
-	WDT_2_29_CYCLES,
-	WDT_2_30_CYCLES,
-	WDT_2_31_CYCLES
-};
+#define WDT_NO_PAUSE_IN_SLEEP	(0)
+#define WDT_PAUSE_IN_SLEEP	BIT(1)
+#define WDT_PAUSE_IN_SLEEP_MASK (1 << 1)
 
+/**
+ * @brief Watchdog behavior when CPU is halted by the debugger.
+ */
+#define WDT_NO_PAUSE_HALTED_BY_DBG	(0)
+#define WDT_PAUSE_HALTED_BY_DBG		BIT(2)
+#define WDT_PAUSE_HALTED_BY_DBG_MASK	(1 << 2)
 
 /**
- * WDT configuration struct.
+ * @brief Watchdog configuration struct.
+ *
+ * timeouts - pointer to an array storing timeouts values in microseconds
+ *	      for all enabled channels. If watchdog supports multiple channels
+ *	      but only one counter reload value then the first value from the
+ *	      array will be taken.
+ * interrupt_fn - function that is going to be called if
+ *		  WDT_OP_MODE_INTERRUPT_RESET is enabled.
+ * operation - a bitfield with following parts:
+ *    operational mode    [ 0 ]       - reset or interrupt -> reset
+ *    pause in sleep      [ 1 ]       - pause in CPU sleep
+ *    pause by debugger   [ 2 ]       - pause when halted by debugger
+ * num_of_channels - number of channels that should be enabled on
+ *		     given watchdog instance.
  */
 struct wdt_config {
-	u32_t timeout;
-	enum wdt_mode mode;
-	void (*interrupt_fn)(struct device *dev);
+	u32_t *timeouts;
+	void (*interrupt_fn)(struct device *dev, u8_t channel);
+	u8_t operation;
+	u8_t num_of_channels;
 };
 
-typedef void (*wdt_api_enable)(struct device *dev);
-typedef void (*wdt_api_disable)(struct device *dev);
+typedef int (*wdt_api_enable)(struct device *dev);
+typedef int (*wdt_api_disable)(struct device *dev);
 typedef int (*wdt_api_set_config)(struct device *dev,
 				  struct wdt_config *config);
 typedef void (*wdt_api_get_config)(struct device *dev,
 				   struct wdt_config *config);
-typedef void (*wdt_api_reload)(struct device *dev);
+typedef void (*wdt_api_reload)(struct device *dev, u8_t channel);
 
 struct wdt_driver_api {
 	wdt_api_enable enable;
 	wdt_api_disable disable;
-	wdt_api_get_config get_config;
 	wdt_api_set_config set_config;
+	wdt_api_get_config get_config;
 	wdt_api_reload reload;
 };
 
-static inline void wdt_enable(struct device *dev)
+/*
+ * @brief Enable watchdog instance.
+ *
```

Comment:
Description of function parameters is missing, add them with @ param. Here and in all the remaining functions.